source/src/molecular-unfolding/cdk/construct-notebook.ts

```diff
@@ -62,60 +67,46 @@ export class Notebook extends Construct {
     this.props = props;
     const INSTANCE_TYPE = 'ml.c5.xlarge';
 
-    this.roleUtil = RoleUtil.newInstance(this, props);
-
-    const githubRepo = 'https://github.com/awslabs/quantum-computing-exploration-for-drug-discovery-on-aws';
+    const srcCodeAsset = new s3_assets.Asset(this, 'srcCodeAsset', {
+      path: path.join(__dirname, '../../'),
+    });
 
-    const defaultCodeRepository = this.node.tryGetContext('default_code_repository') || githubRepo;
+    this.roleUtil = RoleUtil.newInstance(this, props);
 
     const notebookRole = this.roleUtil.createNotebookIamRole();
 
+    srcCodeAsset.bucket.grantRead(notebookRole);
```

Comment:
add asset path to grant least permission

Reply:
changed.